src/wireviz/DataClasses.py

```diff
@@ -2,11 +2,48 @@
 # -*- coding: utf-8 -*-
 
 from typing import Optional, List, Any, Union
-from dataclasses import dataclass, field
-from wireviz.wv_helper import int2tuple
+from dataclasses import dataclass, field, InitVar
+from pathlib import Path
+from wireviz.wv_helper import int2tuple, aspect_ratio
 from wireviz import wv_colors
 
 
+@dataclass
+class Image:
+    gv_dir: InitVar[Path] # Directory of .gv file injected as context during parsing
+    # Attributes of the image object <img>:
+    src: str
+    scale: Optional[str] = None  # false | true | width | height | both
+    # Attributes of the image cell <td> containing the image:
+    width: Optional[int] = None
+    height: Optional[int] = None
+    fixedsize: Optional[bool] = None
+    # Contents of the text cell <td> just below the image cell:
+    caption: Optional[str] = None
+    # See also HTML doc at https://graphviz.org/doc/info/shapes.html#html
+
+    def __post_init__(self, gv_dir):
+
+        if self.fixedsize is None:
+            # Default True if any dimension specified unless self.scale also is specified.
+            self.fixedsize = (self.width or self.height) and self.scale is None
+
+        if self.scale is None:
+            self.scale = "false" if not self.width and not self.height \
+                else     "both"  if     self.width and     self.height \
+                else     "true" # When only one dimension is specified.
+
+        if self.fixedsize:
+            # If only one dimension is specified, compute the other
+            # because Graphviz requires both when fixedsize=True.
+            if self.height:
+                if not self.width:
+                    self.width = self.height * aspect_ratio(gv_dir.joinpath(self.src))
+            else:
+                if self.width:
+                    self.height = self.width / aspect_ratio(gv_dir.joinpath(self.src))
```

Comment:
Perhaps add an `else` below this line:
```py7
else
    raise Exception('fixedsize=True and no dimension is specified!')
```

Reply:
Let's include your suggestion, it's great that you are thinking of all these possible edge cases!

Reply:
This was my suggested addition:
```diff
------------------------- src/wireviz/DataClasses.py --------------------------
index 3ad84d2..e73de9c 100644
@@ -42,6 +42,8 @@ class Image:
             else:
                 if self.width:
                     self.height = self.width / aspect_ratio(gv_dir.joinpath(self.src))
+                else:
+                    raise RuntimeError('fixedsize=True and no dimension is specified!')
 
 
 @dataclass
```
Without this addition, Graphviz does not raise an Exception, as I first thought, but only outputs a warning when this issue occurs (and still generates output files - probably ignoring the invalid attribute value):
```
Warning: fixed cell size with unspecified width or height
in label of node W1
Warning: fixed cell size with unspecified width or height
in label of node W1
```
With the addition, WireViz raises an exception and exit:
```
Traceback (most recent call last):
  File "./wireviz.py", line 257, in <module>
    main()
  File "./wireviz.py", line 253, in main
    parse(yaml_input, file_out=file_out)
  File "./wireviz.py", line 56, in parse
    harness.add_cable(name=key, **attribs)
  File "./../wireviz/Harness.py", line 30, in add_cable
    self.cables[name] = Cable(name, *args, **kwargs)
  File "<string>", line 21, in __init__
  File "./../wireviz/DataClasses.py", line 148, in __post_init__
    self.image = Image(**self.image)
  File "<string>", line 8, in __init__
  File "./../wireviz/DataClasses.py", line 46, in __post_init__
    raise RuntimeError('fixedsize=True and no dimension is specified!')
RuntimeError: fixedsize=True and no dimension is specified!
```
To provoke this issue, the YAML input must contain an image with `fixedsize: True` and no `width` nor `heigth` - which is probably not a simple mistake done by the average user. Are you still sure you prefer the exception before the Graphviz warning? If you prefer the exception, I would suggest to include the connector/cable _name_ (as the wireviz warning does), but that information is currently not available inside the Image dataclass.

Reply:
Hmm, in this case, the GraphViz warning is more helpful, and it clearly specifies which node is causing the problem. You have convinced me that the current implementation is fine, and as you said, this is rather unlikely to happen anyway. Thanks!